openedx_webhooks/lib/edx_repo_tools_data/models.py

```diff
@@ -125,11 +117,18 @@ def agreement_expires_on(self):
         return expires_on
 
     @property
+    def before(self):
+        """
+        Optional[Dict[str, Dict]]: User's `before` data.
+        """
+        return self._data.get('before')
```

Comment:
I was answering your uncertainty about what use cases. We can defer the API design, though I would default to leaving the property private until we know we need it, rather than making it public and then later figuring out if we should take it out.